modules/strawpoll.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+'''
+strawpoll.py - jenni Strawpoll Module
+Copyright 2015, Bekey
+Licensed under the Eiffel Forum License 2.
+
+More info:
+ * jenni: https://github.com/myano/jenni/
+ * Phenny: http://inamidst.com/phenny/
+'''
+
+import urllib2
+import json
+import shlex
+import getopt
+
+def create_strawpoll(jenni, input):
+    """ .strawpoll  --title='Example Title'
+                    [--multi]
+                    [--permissive]
+                    'Option 1'
+                    'Option 2'
+                    'Option 3'
+                    
+    Creates a strawpoll with the given title and options
+    """
+
+    if not input.admin:
+        return;
+    
+    arguments = shlex.split(input)
+    try:
+        optlist, arguments = getopt.getopt(arguments[1:], 't:mp', ['title=', 'multi', 'permissive'])
+    except getopt.GetoptError as err:
+        return jenni.say(str(err))
+    
+    multi = False
+    permissive = False
+    title = "N/A"
+
+    for o, a in optlist:
+        if o in ("-m", "--multi"):
+            multi = True
+        elif o in ("-p", "--permissive"):
+            permissive = True
+        elif o in ("-t", "--title"):
+            title = a
+        else:
+            return jenni.say("Unrecognized options")
+    
+    if len(arguments) == 0:
+        return jenni.say("No options found")
+
+    if len(arguments) > 10:
+        return jenni.say("Too many options, review your input or create a strawpoll manually")
+    
+    data = {'options':arguments,'title':title,'multi':multi,'permissive':permissive}
+    
+    req = urllib2.Request('http://strawpoll.me/api/v2/polls')
+    req.add_header('Content-Type', 'application/json')
+    req.add_header('Referer', '"http://strawpoll.me/"')
+    req.add_header('User-Agent', 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:25.2) Gecko/20150122 Firefox/31.9 PaleMoon/25.2.1')

```

Comment:
The User-Agent was required, otherwise it returned a 403 error. I remember seeing a hardcoded useragent somewhere else, possibly in web.py?

Wasn't sure how to use it, sorry.